### PR TITLE
source.driver optional, target.driver should default based upon target.url

### PR DESCRIPTION
**Pull request by butson**

### CODE REVIEW

File: core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java [hunk 1]
```diff
@@ -289,20 +291,20 @@ protected Group createTargetGroup() {
...
-        group.withOption(driver);
-
         Option url = newBasicOptionBuilder().withName(TARGET_URL)
                 .withDescription(getMessage(TARGET_URL_OPTION_DESCRIPTION)).withRequired(true)
                 .withArgument(newArgumentBuilder().withName(getMessage(TARGET_URL_ARGUMENT_NAME)).withRequired(true)
                         .withMinimum(1).withOptionFormat(optionFormat).build())
                 .build();
         group.withOption(url);
 
+        Option driver = newBasicOptionBuilder().withName(TARGET_DRIVER)
```
Comment (kontaras) [inline core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java:301]: Do you need to add `.withRequired(true)` to the option?

**Review by kontaras — APPROVED** [review]

File: core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java [hunk 2]
```diff
@@ -152,20 +152,22 @@ protected Group createSourceGroup() {
...
                 .withDescription(getMessage(SOURCE_URL_OPTION_DESCRIPTION)).withRequired(true)
                 .withArgument(newArgumentBuilder().withName(getMessage(SOURCE_URL_ARGUMENT_NAME)).withRequired(true)
                         .withMinimum(1).withOptionFormat(optionFormat).build())
                 .build();
         group.withOption(url);
 
+        Option driver = newBasicOptionBuilder().withName(SOURCE_DRIVER)
+                .withDescription(getMessage(SOURCE_DRIVER_OPTION_DESCRIPTION))
+                .withRequired(false)
+                .withArgument(newArgumentBuilder().withName(getMessage(SOURCE_DRIVER_ARGUMENT_NAME)).withRequired(true)
```
Comment (adriansuarez) [inline core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java:165]: I'm confused about the `withRequired(false)... withRequired(true)`. Is it required or not?

Comment (adriansuarez) [reply core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java:165]: It looks like that was actually for the `newArgumentBuilder()`.

**Comment by eresende-nuodb** [comment]
@rshaull @adriansuarez @kontaras What is the status of this PR? Is it ready for merge?

**Comment by adriansuarez** [comment]
@eresende-nuodb: I'm fine with merging these changes.

**Comment by kontaras** [comment]
There was one line that looked wrong to me, but otherwise, I am fine with the code being merged.

File: core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java (hunk 2 above)
Comment (dbutson-mdsol) [reply core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java:165]: newArgumentBuilder returns an Argument or something like that.  so it's setting argument. I'll have to look at the code more.  Welcome to nuodb-migrator which is a lot of code that is inconsistent throughout.

Comment (dbutson-mdsol) [reply core/src/main/java/com/nuodb/migrator/cli/run/CliRunSupport.java:165]: It looks okay.  The option needs to be false, if the option is used then the argument (withArgument) is required.  So the question about withRequired(true) is it is required.   As that is on the Argument not the Option.  The Option is not required.   If you compare before and after only change is the first withRequired().  it was true and is now false.

**Pull request merged**